app/channel.go

```diff
@@ -159,6 +159,27 @@ func (a *App) CreateChannelWithUser(channel *model.Channel, userId string) (*mod
 	return rchannel, nil
 }
 
+// RenameChannel is used to rename the channel Name and the DisplayName fields
+func (a *App) RenameChannel(channel *model.Channel, newChannelName string, newDisplayName string) (*model.Channel, *model.AppError) {
+	var oldChannel *model.Channel
```

Comment:
You are using 2 different styles in the same function. We prefer the later, something like:
```go
oldChannel, err := a.GetChannel(channel.Id)
if err != nil {
	return nil, err
}
```